### DISCUSSION TITLE
Cloning the fork repository through gh does not receive brunches

### DISCUSSION THREAD

**Discussion by ghost** [community/community#84672] [Repositories]
### Select Topic Area

Question

### Body

Help me study and don't understand what is happening
I cloning:
`gh repo clone sunsetshi/PdoOne`
Result of  git branch:
`* master`
I run git branch -a:
```
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/fork
  remotes/origin/master
  remotes/upstream/master
```
fork shows that it is red and not cloned

**Comment by bonizario** [comment]
The `git branch -a` command lists all branches in your repository, both local and remote. Here's what each line in your output means:

- `* master`: This is your current local branch, and the asterisk indicates it's the one you're currently on.
- `remotes/origin/HEAD -> origin/master`: This is a pointer to the default branch on the `origin` remote repository. In this case, it's pointing to `origin/master`.
- `remotes/origin/fork`: This is a branch named `fork` on the `origin` remote repository.
- `remotes/origin/master`: This is the `master` branch on the `origin` remote repository.
- `remotes/upstream/master`: This is the `master` branch on the `upstream` remote repository.

The `fork` branch appearing in red in your terminal is likely due to the color scheme of your terminal and doesn't indicate an issue with the branch itself. It's not cloned because when you clone a repository, by default, you only get a copy of the default branch (usually `master` or `main`).

If you want to work with the `fork` branch, you need to check it out locally with the following command:

```sh
git checkout -b fork origin/fork
```

This command creates a new local branch named `fork`, and sets it up to track the remote branch `origin/fork`. After running this command, you should be able to work with the `fork` branch locally.

If this answer helped you, please remember to mark it as the "accepted answer" 🚀.
